docs/developer-docs/latest/setup-deployment-guides/typescript.md

```diff
@@ -81,9 +81,39 @@ yarn develop
 
 </code-group>
 
-<!--## Use TypeScript typings
+## Use TypeScript typings
+
+Strapi provides typings on the Strapi class to improve the TypeScript experience. In the following example a `register` method is added to `./src/index.ts`. For example, when the `strapi.runLifecyclesFunctions` method is added, a list of available lifecycle types (`register`, `bootstrap` and `destroy`) are returned by the code editor. The following example is illustrated statically and the second tab shows an animated example.
+
+:::: tabs card
+
+::: tab STATIC EXAMPLE
+
+```ts
+// path: ./src/index.ts
+
+import '@strapi/strapi';
+
+export default {
+
+  register( {strapi }: { strapi: Strapi }) {
+    strapi.runLifecyclesFunctions('register') .then(()  => console.log('hello world')
+    )
+    
+  },
+};
+```
+
+:::
+
+::: tab ANIMATED GIF
+
+![Animated gif demonstrating TypeScript typings](./TypeScript-typing.gif)
+
+:::
```

Comment:
I added the gif as an example because it was requested multiple times by the @Convly  and @marcoautiero. We are having an active conversation about how to best communicate typings. As far as I am aware there are multiple tabs tables throughout the documentation with code examples, which is exactly what this is.

Reply:
> As far as I am aware there are multiple tabs tables throughout the documentation with code examples, which is exactly what this is.

This is not true, Shaun. Tabs are used when we have several options for the same context, but each tab with a different purpose, _and_ it would be too long to have the same content stacked up from top to bottom. The [relations](https://docs.strapi.io/developer-docs/latest/development/backend-customization/models.html#relations) subsection is a good example of the usage of tabs, for instance.
VuePress code groups are what we use for various examples of codes all having the same purpose. 
But we have zero examples of tabs used like here. The Quick Start Guide is the only place that features GIFs because it actually is a tutorial. And even in this case, the GIF is above or below the content, allowing the reader to see both versions (text and image) and choose what they want to see, or even follow both simultaneously.